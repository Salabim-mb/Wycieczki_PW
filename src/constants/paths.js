export const paths = {
	DASHBOARD: {
		name: "Strona główna",
		path: "/",
	},
	BLOG: {
		name: 'Blog',
		path: '/:blog',
	},
	BLOG_CATEGORIES: {
		name: "Kategorie",
		path: "/blog/:category",
		redirect: (newCategory) => `/blog/${newCategory}`,
	},
	BLOG_ENTRY: {
		name: "Wpis",
		path: "/blog/:category/:entry_id",
		redirect: (category, entryId) => `/blog/${category}/${entryId}`,
	},
	POST_CREATOR: {
		name: 'Kreator wpisów',
		path: '/blog/post-creator/:id?',
		redirect: (entryId) => `/blog/post-creator/${entryId}`
	},
	CREATOR_CATEGORY: {
		name: "Krator kategorii",
		path: "/blog/category-creator/:id?",
		redirect: (categoryId) => `/blog/category-creator/${categoryId}`
	}
}