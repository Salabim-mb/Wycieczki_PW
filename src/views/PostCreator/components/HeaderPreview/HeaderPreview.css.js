import styled from 'styled-components';

export const StyledImage = styled.img`
	max-width: 100%;
`;
