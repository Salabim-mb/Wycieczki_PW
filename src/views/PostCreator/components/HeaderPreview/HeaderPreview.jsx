import React, { useMemo } from 'react';
import { Accordion, AccordionSummary, AccordionDetails, Typography } from '@material-ui/core';
import { ExpandMore } from '@material-ui/icons';
import { isYoutubeLink } from 'utils/helpers';
import { StyledImage } from './HeaderPreview.css';

const HeaderPreview = ({ header }) => {
	const wait = useMemo(() => {});
	console.log(wait, header);
	return (
		<Accordion>
			<AccordionSummary
				expandIcon={<ExpandMore />}
				aria-controls="panel1a-content"
				id="panel1a-header"
			>
				<Typography>Podgląd nagłówka posta</Typography>
			</AccordionSummary>
			<AccordionDetails>
				{header ? (
					isYoutubeLink(header) ? null : (
						<StyledImage src={typeof header === 'string' ? header : URL.createObjectURL(header)} />
					)
				) : null}
			</AccordionDetails>
		</Accordion>
	);
};

export default HeaderPreview;
