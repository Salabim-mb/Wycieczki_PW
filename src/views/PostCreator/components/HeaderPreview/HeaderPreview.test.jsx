import React from 'react';
import { render } from '@testing-library/react';
import paths from 'constants/api';
import HeaderPreview from './HeaderPreview';

describe('HeaderPreview', () => {
	let props;

	beforeEach(() => {
		props = {
			header: paths.PLAIN,
		};
	});

	it('should match snapshot', () => {
		const { container } = render(<HeaderPreview {...props} />);

		expect(container).toMatchSnapshot();
	});

	it('should match youtube link', () => {
		props.header = 'https://www.youtube.com/xdd';
		const { container } = render(<HeaderPreview {...props} />);

		expect(container).toMatchSnapshot();
	});

	it('should match file', () => {
		props.header = new File(['foo'], 'foo.jpg', {
			type: 'image/jpg',
		});

		global.URL.createObjectURL = jest.fn(() => '/link');

		const { getByTestId } = render(<HeaderPreview {...props} />);

		expect(getByTestId('header')).toBeInTheDocument();
	});

	it('should not render header if it not exist', () => {
		props.header = '';

		const { queryByTestId } = render(<HeaderPreview {...props} />);

		expect(queryByTestId('header')).not.toBeInTheDocument();
	});

	it('should not render header if someone already typing youtube link', () => {
		props.header = 'abc';

		const { queryByTestId } = render(<HeaderPreview {...props} />);

		expect(queryByTestId('header')).not.toBeInTheDocument();
	});
});
