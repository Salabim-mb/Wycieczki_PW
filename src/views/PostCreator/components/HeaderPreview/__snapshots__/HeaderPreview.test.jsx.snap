// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderPreview should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-controls="panel1a-content"
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root"
      id="panel1a-header"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Podgląd nagłówka posta
        </p>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-hidden"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="panel1a-header"
            id="panel1a-content"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root"
            >
              <img
                class="sc-bdfBwQ hBgaUH"
                data-testid="header"
                src="https://wut-vtrips.herokuapp.com"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HeaderPreview should match youtube link 1`] = `
<div>
  <div
    class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-controls="panel1a-content"
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root"
      id="panel1a-header"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Podgląd nagłówka posta
        </p>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-hidden"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="panel1a-header"
            id="panel1a-content"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root"
            >
              https://www.youtube.com/xdd
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
